src/main/java/org/spongepowered/common/conversation/SpongeConversationArchetype.java

```diff
@@ -0,0 +1,166 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.conversation;
+
+import com.google.common.collect.ImmutableSet;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.conversation.Conversant;
+import org.spongepowered.api.conversation.Conversation;
+import org.spongepowered.api.conversation.ConversationArchetype;
+import org.spongepowered.api.conversation.EndingHandler;
+import org.spongepowered.api.conversation.ExternalChatHandler;
+import org.spongepowered.api.conversation.Question;
+import org.spongepowered.api.plugin.PluginContainer;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.common.service.pagination.PaginationCalculator;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class SpongeConversationArchetype implements ConversationArchetype {
+
+    private final Question question;
+    private Set<EndingHandler> endingHandlers = new HashSet<>();
+    private final ExternalChatHandler defaultChatHandler;
+    private final String id;
+    private final String exit;
+    @Nullable private final Text header;
+    @Nullable private final Text title;
+    @Nullable private final Text padding;
+    @Nullable private final Text startingMessage;
+    private boolean catchesOutput = true;
+    private boolean allowCommands = false;
+
+    /**
+     * Creates the conversation archetype. Generally should only be called from
+     * {@link SpongeConversationArchetypeBuilder}.
+     *
+     * @param firstQuestion The first question
+     * @param catchesOutput Whether or not to catch conversant output
+     * @param defaultHandler The default handler applied to conversants
+     * @param endingHandlers The ending handlers for the conversation
+     * @param startingMessage The message sent to conversants at the start of
+     *     the conversation
+     * @param id The id of the archetype
+     * @param exit The exit keyword
+     * @param title The title of the conversation
+     * @param padding The padding for the title
+     */
+    SpongeConversationArchetype(Question firstQuestion, boolean catchesOutput, boolean allowCommands, ExternalChatHandler defaultHandler,
+        Set<EndingHandler> endingHandlers, @Nullable Text startingMessage, String id, String exit, @Nullable Text title, @Nullable Text padding) {
+        this.question = firstQuestion;
+        this.endingHandlers = endingHandlers;
+        this.catchesOutput = catchesOutput;
+        this.allowCommands = allowCommands;
+        this.defaultChatHandler = defaultHandler;
+        this.startingMessage = startingMessage;
+        this.id = id.toLowerCase();
+        this.exit = exit.toLowerCase();
+        this.title = title;
+        this.padding = padding;
+        if (title != null && padding != null) {
```

Comment:
A header is generally composed of a title and it's padding like pagination but I like this idea. If a header is set then that would override those so it could be whatever they so desire.